packages/graph/hash_graph/lib/graph/src/ontology/property_type.rs

```diff
@@ -26,7 +31,243 @@ impl QueryRecord for PropertyType {
 impl<'q> TryFrom<Path> for PropertyTypeQueryPath<'q> {
     type Error = Report<de::value::Error>;
 
-    fn try_from(_path: Path) -> Result<Self, Self::Error> {
-        todo!("https://app.asana.com/0/1203007126736607/1203167266370354/f")
+    fn try_from(path: Path) -> Result<Self, Self::Error> {
+        Self::deserialize(de::value::SeqDeserializer::new(
+            path.segments.into_iter().map(|segment| segment.identifier),
+        ))
+        .into_report()
+    }
+}
+
+/// A single token in a [`DataTypeQueryPath`].
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum PropertyTypeQueryToken {
```

Comment:
So we now have `PropertyTypeQueryToken`, `PropertyTypeQueryPath`, _and_ `PropertyTypeField`?

I appreciate we sometimes just can't get around this verbosity, so I'm not expecting changes, but do we *really* need all 3? They're so similar and it seems like it'll just be a pain to update anything.

Reply:
`Path` and `Field` is required (as pointed out above). `Token` is not strictly required, but it's easier to use that enum instead of plain strings. Without `Token`, the `visit_seq` could be rewritten as 
```rust
fn visit_seq<A>(mut self, mut seq: A) -> Result<Self::Value, A::Error>
where
    A: SeqAccess<'de>,
{
    let token: String = seq
        .next_element()?
        .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
    self.position += 1;
    Ok(match token.as_str() {
        "ownedById" => PropertyTypeQueryPath::OwnedById,
        "baseUri" => PropertyTypeQueryPath::BaseUri,
        "versionedUri" => PropertyTypeQueryPath::VersionedUri,
        "version" => PropertyTypeQueryPath::Version,
        "title" => PropertyTypeQueryPath::Title,
        "description" => PropertyTypeQueryPath::Description,
        "dataTypes" => {
            seq.next_element::<Selector>()?
                .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
            self.position += 1;

            let data_type_query_path =
                DataTypeQueryPathVisitor::new(self.position).visit_seq(seq)?;

            PropertyTypeQueryPath::DataTypes(data_type_query_path)
        }
        "propertyTypes" => {
            seq.next_element::<Selector>()?
                .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
            self.position += 1;

            let property_type_query_path = Self::new(self.position).visit_seq(seq)?;

            PropertyTypeQueryPath::PropertyTypes(Box::new(property_type_query_path))
        }
        unexpected => return Err(de::Error::invalid_value(Unexpected::Str(unexpected), &self)),
    })
}
```

but this requires deserializing into `String`. Using `Token` is easier and probably faster.

Reply:
Understood, just would be nice for us not to have such duplication.

Reply:
I agree, but currently, I don't know a good way around this. 🙁